datalad_next/runners/run.py

```diff
@@ -0,0 +1,166 @@
+"""
+This module provides a context-manager that executes a subprocess.
+
+The context manager supports the termination and/or killing of the subprocess,
+if it is executed with a not-None `terminate_time` and/or `kill_time`.
+
+The context manager supports generator- and non-generator-protocols.
+
+"""
+from __future__ import annotations
+
+import subprocess
+from collections.abc import Generator
+from contextlib import contextmanager
+from pathlib import Path
+from queue import Queue
+from subprocess import DEVNULL
+from typing import (
+    Any,
+    IO,
+)
+
+from datalad_next.runners import (
+    GeneratorMixIn,
+    Protocol,
+    ThreadedRunner,
+)
+
+
+def create_kill_wrapper(cls: type[Protocol]) -> type[Protocol]:
```

Comment:
Looks like this is an internal helper

```suggestion
def _create_kill_wrapper(cls: type[Protocol]) -> type[Protocol]:
```

Reply:
Done